### ISSUE TITLE
chef.upgrade folder not deleted on w2k12r2 and w2k16 after upgrade

### ISSUE THREAD

**Issue by jdgoins** [chef-cookbooks/chef_client_updater#70]
### Cookbook version
3.0.4

### Chef-client version
13.5.3

### Platform Details
w2k12r2 or w2k16

### Scenario:
This cookbook keeps trying to delete the chef.upgrade folder in C:\opscode after a successful upgrade. It does not error out, but it fails to remove the chef.upgrade folder.

### Steps to Reproduce:
1. Upgrade from Chef Client 12.19.36 to 13.5.3 using this cookbook (3.0.4) on w2k12r2.
```
chef_client_updater 'Enforce_chef_client_version' do
  version '13.5.3'
  post_install_action 'kill'
end
```

2. Run chef-client again. Subsequent runs will fail to remove the chef.upgrade folder.

### Expected Result:
- Cookbook should be idempotent. 
- Cookbook should delete chef.upgrade folder after upgrade is completed.
- Cookbook should error if it fails to delete chef.upgrade folder. 

### Actual Result:
- Cookbook is not idempotent after upgrade.
- Cookbook does not delete chef.upgrade folder after upgrade is completed. 
- Cookbook does not error when it fails to delete chef.upgrade folder.

#### Debug Output For Runs After Upgrade
```
  * chef_client_updater[Enforce_chef_client_version] action update[2017-10-25T14:42:51-06:00] INFO: Processing chef_clie
nt_updater[Enforce_chef_client_version] action update (test_cookbook::default line 13)
[2017-10-25T14:42:51-06:00] DEBUG: Providers for generic chef_client_updater resource enabled on node include: [LWRP pro
vider chef_client_updater from cookbook chef_client_updater]
[2017-10-25T14:42:51-06:00] DEBUG: Provider for action update on resource chef_client_updater[Enforce_chef_client_versio
n] is LWRP provider chef_client_updater from cookbook chef_client_updater

    - remove c:/opscode/chef.upgrade from previous chef-client run2.6.13
[2017-10-25T14:42:52-06:00] DEBUG: Found gem version 2.6.13. Desired version is at least 2.6.11
[2017-10-25T14:42:53-06:00] DEBUG: Platform detected as {:platform=>"windows", :platform_version=>"2012r2", :architectur
e=>"x86_64"} by mixlib_install
[2017-10-25T14:42:53-06:00] DEBUG: Passing options to mixlib-install: {:product_name=>"chef", :platform_version_compatib
ility_mode=>true, :platform=>"windows", :platform_version=>"2012r2", :architecture=>"x86_64", :channel=>:stable, :produc
t_version=>"13.5.3"}
[2017-10-25T14:42:53-06:00] DEBUG: Desired version in specified channel maps to 13.5.3
[2017-10-25T14:42:53-06:00] DEBUG: The current chef-client version is 13.5.3 and the desired version is 13.5.3
```

**Comment by tas50** [comment]
We have some changes to the cleanup in the pipeline that *may* resolve some of the random issues with failures and cleanup on windows. Soon!

**Comment by tas50** [comment]
I've released 3.1.0 which reworks the cleanup logic. If you're still seeing this issue with that version feel free to open this issue back up
